catalystwan/typed_list.py

```diff
@@ -259,3 +259,20 @@ def first(self) -> T:
             raise InvalidOperationError("The input sequence contains no elements.")
 
         return self.data[0]
+
+    def find(self, **kwargs) -> T:
```

Comment:
Method `find` should return None instead raise an exception in my opinion. Method get() could raise exception.

Reply:
The point was to have `T` as return annotation not `Optional[T]`. `find()` should be used when exactly one element is expected - otherwise use `filter()`